packages/flutter/test/material/selection_area_test.dart

```diff
@@ -130,6 +130,50 @@ void main() {
     variant: const TargetPlatformVariant(<TargetPlatform>{TargetPlatform.iOS}),
   );
 
+  testWidgets(
+    'builds the default context menu by default on Android and iOS web',
+    (WidgetTester tester) async {
+      final FocusNode focusNode = FocusNode();
+      addTearDown(focusNode.dispose);
+
+      await tester.pumpWidget(
+        MaterialApp(
+          home: SelectionArea(focusNode: focusNode, child: const Text('How are you?')),
+        ),
+      );
+      await tester.pumpAndSettle();
+
+      expect(find.byType(AdaptiveTextSelectionToolbar), findsNothing);
+
+      // Show the toolbar by longpressing.
+      final RenderParagraph paragraph1 = tester.renderObject<RenderParagraph>(
+        find.descendant(of: find.text('How are you?'), matching: find.byType(RichText)),
+      );
+      final TestGesture gesture = await tester.startGesture(
+        textOffsetToPosition(paragraph1, 6),
+      ); // at the 'r'
+      addTearDown(gesture.removePointer);
+      await tester.pump(const Duration(milliseconds: 500));
+      // `are` is selected.
+      expect(paragraph1.selections[0], const TextSelection(baseOffset: 4, extentOffset: 7));
+
+      await gesture.up();
+      await tester.pumpAndSettle();
+
+      expect(find.byType(AdaptiveTextSelectionToolbar), findsOneWidget);
+    },
+    // TODO(Renzo-Olivares): Remove this test when the web context menu
+    // for Android and iOS is re-enabled.
+    // See: https://github.com/flutter/flutter/issues/177123.
+    // [intended] Android and iOS use the flutter rendered menu on the web.
+    skip:
+        !kIsWeb ||
+        !<TargetPlatform>{
+          TargetPlatform.android,
+          TargetPlatform.iOS,
+        }.contains(defaultTargetPlatform),
+  );
```

Comment:
Is there an easy way to add a similar test to the `SelectableRegion` layer too? I don't feel strongly about this, feel free to skip (or add this test in a follow-up PR).

Reply:
Awesome! :D